### ISSUE TITLE
YOLO World LVIS Validation Accuracy

### ISSUE THREAD

**Issue by erictsai1224tw** [ultralytics/ultralytics#17901]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi Ultralytics developers,

Thanks for providing such fine yolo-based computer vision package!
Recently, I am conducting some experiments about YOLO World. Unfortunately, I only got about 5% accuracy on mAP50:95.
My checkpoint is "yolov8s-world.pt" from Ultralytics, and I am not sure which part I may not do correct.
 
My evaluation code is shown bellow, please give me some guide about which part i may do wrong. Thanks for your kindly help!
`=python

  project = "validation/yoloworld/lvis"
  name = "test1"
  
  ckpt_name = 'yolov8s-world.pt'
  model = YOLOWorld("yolov8s-world.yaml")
  model._load(ckpt_name, task='task')
  with open("lvis_categories.json") as f:
    lvis_class = json.load(f) # This will load all categories in LVIS dataset to list
  model.set_classes(lvis_class)
  
  metrics = model.val(data="lvis.yaml",
                      batch=16,
                      plots = True,
                      project=project,
                      name=name
                      )
  
  metrics_dict = {
    "map50_95": metrics.box.map.tolist(),
    "map50": metrics.box.map50.tolist(),
    "map75": metrics.box.map75.tolist(),
    "map_all_class": metrics.box.maps.tolist()
  }
`

**Comment by UltralyticsAssistant** [comment]
👋 Hello @erictsai1224tw, thank you for your interest in Ultralytics 🚀! We appreciate you taking the time to share your question and include details about your evaluation and validation workflow.

If this is a 🐛 Bug Report, we kindly request a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist with debugging. This should include a clear dataset sample, related code (as you helpfully shared), expected behavior, and observed behavior.

If this is more of a custom training or model evaluation ❓ Question, please ensure the following:
1. Double-check your `yolov8s-world.pt` checkpoint compatibility with your dataset and configuration, and refer to our [model documentation](https://docs.ultralytics.com/models/yolov8/).
2. Share further details about your experiment setup such as dataset splits, preprocessing steps, or any modifications made to the original LVIS dataset configuration.
3. Confirm that you are following [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), especially ensuring that the dataset, classes, and evaluation metrics are correctly specified.
4. Consider adding any relevant validation logs, warnings, or errors if applicable.

Feel free to join the Ultralytics community for further support and knowledge sharing:
- Real-time chat? Join our [Discord](https://discord.com/invite/ultralytics) server 🎧.
- Detailed discussions? Head over to [Discourse](https://community.ultralytics.com).
- Want to explore or share ideas? Visit our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Ensure you are using the latest `ultralytics` package release by upgrading your version in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). You can do so with this command:

```bash
pip install -U ultralytics
```

After upgrading, retry your validation experiment to see if the behavior improves.

## Test in Verified Environments

YOLO models, including newly developed ones like `YOLOWorld`, work well in our pre-verified computation environments, which may help eliminate platform-related issues:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to guide you toward quickly resolving your issue, but no worries – an Ultralytics engineer will follow up with you soon to provide further assistance! 😊

**Comment by Y-T-G** [comment]
What's inside `lvis_categories.json`?

**Comment by Y-T-G** [comment]
I guess you need to create a custom `lvis.yaml` where you select one of the names when there multiple names for the same class separated by `/`

**Comment by erictsai1224tw** [comment]
Hi,

The file `lvis_categories.json` contains the list of LVIS's class name, just like:
[
    "aerosol_can",
    "air_conditioner",
    "airplane",
    "alarm_clock",
    "alcohol",
    "alligator",
    "almond",
    "ambulance",
    "amplifier",
    "anklet",
    "antenna",
    "apple",
    "applesauce",
    ......
]
The purpose of this list is for the set class function of YOLO World.

My lvis.yaml is also provided by ultralytics, it is like

```python3
# Ultralytics YOLO 🚀, AGPL-3.0 license
# LVIS dataset http://www.lvisdataset.org by Facebook AI Research.
# Documentation: https://docs.ultralytics.com/datasets/detect/lvis/
# Example usage: yolo train data=lvis.yaml
# parent
# ├── ultralytics
# └── datasets
#     └── lvis  ← downloads here (20.1 GB)

# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: lvis # dataset root dir
# train: train.txt # train images (relative to 'path') 100170 images
# val: val.txt # val images (relative to 'path') 19809 images
train: train.txt
val: val.txt
minival: minival.txt # minval images (relative to 'path') 5000 images

names:
  0: aerosol can/spray can
  1: air conditioner
  2: airplane/aeroplane
  3: alarm clock
  4: alcohol/alcoholic beverage
  5: alligator/gator
  6: almond
  7: ambulance
  8: amplifier
  9: anklet/ankle bracelet
  10: antenna/aerial/transmitting aerial
  11: apple
  ...
```

**Comment by Y-T-G** [comment]
Yeah, you need names in yaml to be similar to how it is JSON. The names in the yaml have `/` with synonyms.

**Comment by erictsai1224tw** [comment]
Thanks for your mention, I did not see them are different before!
After I fixed this issue, the mAP50:95 has improved a little, still at 5%.
Is there any other issue made this happen?
Thanks lot again for your help!

**Comment by Y-T-G** [comment]
What did you change? The yaml names?

**Comment by erictsai1224tw** [comment]
Yeah, I ensure the name list put into function`set class` same as the name in the yaml

**Comment by Y-T-G** [comment]
Change the names in the yaml, not the JSON. The JSON is correct. The one is yaml isn't because it has synonyms.

**Comment by KaamComputingNp** [comment]
@erictsai1224tw Hi, were you able to reproduce the LVIS Validation Accuracy after changing the names in YAML?
